bin/functions

```diff
@@ -47,6 +47,8 @@ function check-jq() {
 }
 
 function retrieve-adoptopenjdk() {
+  local -a cache_files cache_timestamps
```

Comment:
```suggestion
```
Will declare closer to usage

Reply:
I'm unsure about this. In many languages I do like declaring things as close as possible to where they are used. This doesn't feel idiomatic in shell though.